bottom-navigator/src/main/java/com/pandora/bottomnavigator/BottomNavigator.kt

```diff
@@ -129,16 +131,43 @@ open class BottomNavigator internal constructor() : ViewModel() {
      * Ideally you desing your fragments to be able to handle detach/attach but in some situations
      * that might not be feasable. By setting detachable = false the fragment's view will be kept
      * and memory and just hidden view.
+     *
+     * @param enterAnim An animation or animator resource ID used for the enter animation on the
+     *              view of this fragment when it's added.
+     * @param exitAnim An animation or animator resource ID used for the exit animation on the
+     *              view of the fragment being removed or detached when this fragment is added.
+     * @param popEnterAnim An animation or animator resource ID used for the enter animation on the
+     *              view of the resuming fragment when it is re-added or reattached when this
+     *              fragment is popped
+     * @param popExitAnim An animation or animator resource ID used for the exit animation on this
+     *              fragment when it is popped
      */
-    open fun addFragment(fragment: Fragment, detachable: Boolean = true) {
-        addFragmentInternal(fragment, currentTab, detachable)
+    open fun addFragment(
+        fragment: Fragment,
+        detachable: Boolean = true,
+        @AnimatorRes @AnimRes enterAnim: Int = 0,
+        @AnimatorRes @AnimRes exitAnim: Int = 0,
+        @AnimatorRes @AnimRes popEnterAnim: Int = 0,
+        @AnimatorRes @AnimRes popExitAnim: Int = 0
```

Comment:
Now that I think about it these 0 default values are going to cause the animation to be added to all fragments. Would it be better to make these nullable with a default value of null so the FragmentTransactionHandler can skip adding an animation at all? Or maybe it's better have a container data class for the 4 values. You already have one with TransitionsData, maybe we could make it part of the public interface here and make it nullable. I'm not sure, just kind of musing aloud here, what do you think?

Reply:
We could leave the existing `addFragment(fragment: Fragment, detachable: Boolean = true)` and add a new `addFragment` method with the animations, so TransitionsData is null if the anim params aren't used

Reply:
Ok let's do that.

Reply:
Updated 👍